libraries/chain/controller.cpp

```diff
@@ -3680,13 +3694,23 @@ block_state_legacy_ptr controller::create_block_state( const block_id_type& id,
 
 void controller::push_block( controller::block_report& br,
                              const block_state_legacy_ptr& bsp,
+                             const forked_branch_callback_legacy& forked_branch_cb,
+                             const trx_meta_cache_lookup& trx_lookup )
+{
+   validate_db_available_size();
+   my->push_block( br, bsp, forked_branch_cb, trx_lookup );
+}
+
+void controller::push_block( controller::block_report& br,
+                             const block_state_ptr& bsp,
```

Comment:
Don't think we will want to public push_block like this. Users of controller shouldn't need to know which one to call.

Reply:
Sure, I agree and ideally the `block_state` struct should not be visible outside `libraries/chain`. But I thought that if `create_block_state` and `create_block_state_future` were needed maybe `push_block` as well? We can certainly remove these if not needed.

Reply:
Worse case we have some type that controller provides via something like create_block_start_state() that `controller::push_block` breaks apart and calls the appropriate `controller_impl::push_block`.

Reply:
This is fine for now until we get the create_block_state() and create_block_state_future() fixed.